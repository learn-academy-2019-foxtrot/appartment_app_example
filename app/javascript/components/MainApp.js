import React from "react"
import PropTypes from "prop-types"
class MainApp extends React.Component {
  render () {
    const{ signed_in, sign_in_route, sign_out_route } = this.props
    return (
      <React.Fragment>
        <h1>Appartment App</h1>
        {signed_in &&
          <a href={sign_out_route}>Sign Out</a>
        }

        {!signed_in &&
          <a href={sign_in_route}>Sign In</a>
        }
      </React.Fragment>
    );
  }
}

export default MainApp
