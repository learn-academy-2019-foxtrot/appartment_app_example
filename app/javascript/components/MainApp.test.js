import React from 'react';
import renderer from 'react-test-renderer';

import MainApp from './MainApp'

it('renders for a signed out user', () => {
  const tree = renderer
    .create(<MainApp signed_in={false} sign_in_route="/users/sign_in" sign_out_route="/users/sign_out" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders for a signed in user', () => {
  const tree = renderer
    .create(<MainApp signed_in={true} sign_in_route="/users/sign_in" sign_out_route="/users/sign_out" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
