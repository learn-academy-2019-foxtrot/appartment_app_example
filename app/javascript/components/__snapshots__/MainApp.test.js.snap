// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders for a signed in user 1`] = `
Array [
  <h1>
    Appartment App
  </h1>,
  <a
    href="/users/sign_out"
  >
    Sign Out
  </a>,
]
`;

exports[`renders for a signed out user 1`] = `
Array [
  <h1>
    Appartment App
  </h1>,
  <a
    href="/users/sign_in"
  >
    Sign In
  </a>,
]
`;
